### PR TITLE
feat(loader): upgrade purgecss version

### PR DESCRIPTION
**Pull request by infoxicator**
***This is a non-breaking change for consumers***

- Upgraded `purgecss` version to 2.1.2

- Implemented the following additional options as described in the purgecss options documentation https://purgecss.com/configuration.html#options

1. `fontFace`
2. `keyFrames`
3. `variables`
4. `whitelist`
5. `whitelistPatters`
6. `whitelistPatternsChildren`

- fixes https://github.com/americanexpress/purgecss-loader/issues/3

**TESTING:**

Tested doing `npm pack` and installing this version in `one-app-bundler`, then `npm pack` again and installed the updated bundler in one of the docs repositories, confirming the changes indeed allow `:global{}` classes when `purgecss loader` is enabled in the bundler configuration and it removes unused classes even if wrapped in `:global{}`

### CODE REVIEW

File: __fixtures__/root.scss [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (c) 2018 American Express Travel Related Services Company, Inc.
```
Comment (nellyk) [inline __fixtures__/root.scss:2]: not sure if this date matters?

Comment (infoxicator) [reply __fixtures__/root.scss:2]: @nellyk happy to change it, do we change it on all files or just the new ones I created?

Comment (10xLaCroixDrinker) [reply __fixtures__/root.scss:2]: Just new ones

Comment (infoxicator) [reply __fixtures__/root.scss]: fixed on https://github.com/americanexpress/purgecss-loader/pull/9/commits/bddff983e899cfac17c8239e690a15db6a33351b

File: __tests__/loader.spec.js [hunk 2]
```diff
@@ -96,4 +97,19 @@ describe('purgecss loader', () => {
         expect(output).toContain('isUsed');
       });
   });
+
+  it('should not strip global classes', () => {
```
Comment (JAdshead) [inline __tests__/loader.spec.js:101]: this can be a async function, then you can drop the use of `expect.assertions(2);` and use await

Comment (infoxicator) [reply __tests__/loader.spec.js]: fixed in https://github.com/americanexpress/purgecss-loader/pull/9/commits/3fa1871140eb4d07fb7474db15a3966b1390264a 👍

**Comment by nellyk** [comment]
Could we please hold off on merging this until we've added semantic release #10

**Pull request merged**